posthog/temporal/data_imports/pipelines/snowflake/snowflake.py

```diff
@@ -0,0 +1,150 @@
+from typing import Any, Optional
+from collections.abc import Iterator
+
+from posthog.temporal.common.logger import FilteringBoundLogger
+from posthog.temporal.data_imports.pipelines.helpers import incremental_type_to_initial_value
+from posthog.temporal.data_imports.pipelines.pipeline.typings import SourceResponse
+from posthog.warehouse.types import IncrementalFieldType
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from dlt.common.normalizers.naming.snake_case import NamingConvention
+import snowflake.connector
+from snowflake.connector.cursor import SnowflakeCursor
+
+
+def _get_connection(
+    account_id: str,
+    user: Optional[str],
+    password: Optional[str],
+    passphrase: Optional[str],
+    private_key: Optional[str],
```

Comment:
I think you could possibly handle the optionals earlier in the other function, to avoid optional parameter passing through 2 functions the chain here, but not a huge deal either way.